sqlglot/dialects/duckdb.py

```diff
@@ -41,6 +44,13 @@ def _ts_or_ds_add_sql(self: DuckDB.Generator, expression: exp.TsOrDsAdd) -> str:
 def _date_delta_sql(
     self: DuckDB.Generator, expression: exp.DateAdd | exp.DateSub | exp.TimeAdd
 ) -> str:
+    if isinstance(expression, exp.DateSub):
+        subexpr = expression.expression
+        if not isinstance(subexpr, exp.Literal) or not subexpr.is_int:
+            # Preserve the DATE_SUB roundtrip; If subexpr was an integer, we'd
+            # convert to interval as in below (e.g. BigQuery -> DuckDB transpilation)
+            return self.func("DATE_SUB", f"'{expression.unit}'", subexpr, expression.this)
```

Comment:
Can't we just always generate the datesub syntax?

Reply:
Discussed in Slack - there's a concern related to `INTERVAL x <unit>`, ie. the expression is a column.